### PR TITLE
Openstack delete dynamic floating ip in delete cluster 

### PR DESCRIPTION
**Pull request by drekle**
Instead of using the cluster name to search for the appropriate loadbalancers, which limits this to the api loadbalancer provisioned by kops, I have updated this to search for loadbalancers on the subnets which kops provisions which will clean up the resources created by LoadBalancer services internal to the cluster.

Without the change `kops delete cluster` will fail attempting to delete a subnet with associated ports.

/sig openstack

### CODE REVIEW

File: pkg/resources/openstack/lb.go [hunk 1]
```diff
@@ -17,27 +17,32 @@ limitations under the License.
...
+	typeLBL  = "LBListener"
+	typeLBP  = "LBPool"
+	typeLBPM = "LBPoolMonitor"
 )
 
-func (os *clusterDiscoveryOS) ListLB() ([]*resources.Resource, error) {
+func (os *clusterDiscoveryOS) DeleteSubnetLBs(subnet subnets.Subnet) ([]*resources.Resource, error) {
 	var resourceTrackers []*resources.Resource
+
+	if os.osCloud.UseOctavia() {
```
Comment (drekle) [inline pkg/resources/openstack/lb.go]: @zetaab I am not overly familiar with octavia if you could give this a look.

Comment (zetaab) [reply pkg/resources/openstack/lb.go]: if using octavia we do not need delete any subresources (listener,monitor,pool,members). Instead we can just delete loadbalancer itself and use cascade option there. So most part of the code is useless if using octavia. Example when using octavia https://github.com/kubernetes/kops/blob/master/pkg/resources/openstack/lb.go#L53-L56

Comment (zetaab) [inline pkg/resources/openstack/lb.go]: remove this whole if structure

File: pkg/resources/openstack/lb.go [hunk 2]
```diff
@@ -57,25 +62,28 @@ func (os *clusterDiscoveryOS) ListLB() ([]*resources.Resource, error) {
 			},
 		}
 		resourceTrackers = append(resourceTrackers, resourceTracker)
-	}
-	return resourceTrackers, nil
-}
 
-func (os *clusterDiscoveryOS) ListLBPools() ([]*resources.Resource, error) {
-	var resourceTrackers []*resources.Resource
+		//Identify pools associated to this LB
```
Comment (zetaab) [inline pkg/resources/openstack/lb.go]: octavia does not need any of these for loops (listeners, pools or monitors) so add if here something like execute only `if !os.osCloud.UseOctavia()`

File: pkg/resources/openstack/lb.go (hunk 1 above)
Comment (drekle) [reply pkg/resources/openstack/lb.go]: Thanks ill update to reflect this.

**Comment by drekle** [comment]
There may be some ports associated to the loadbalancer which will not be cleaned up.

**Comment by drekle** [comment]
@zetaab I have not changed ports to query all ports on the cluster network, and then delete them.  Previously this was trying to grep by name.  There was an empty name for dynamically created ports.

**Pull request merged**